pkg/cmd/admin/policy/who_can.go

```diff
@@ -69,12 +72,23 @@ func (o *whoCanOptions) run() error {
 	resourceAccessReview.Resource = o.resource
 	resourceAccessReview.Verb = o.verb
 
-	resourceAccessReviewResponse, err := o.client.ResourceAccessReviews(o.bindingNamespace).Create(resourceAccessReview)
+	var reviewInterface client.ResourceAccessReviewInterface
+	if o.allNamespaces {
+		reviewInterface = o.client.ClusterResourceAccessReviews()
+	} else {
+		reviewInterface = o.client.ResourceAccessReviews(o.bindingNamespace)
+	}
+
+	resourceAccessReviewResponse, err := reviewInterface.Create(resourceAccessReview)
 	if err != nil {
 		return err
 	}
 
-	fmt.Printf("Namespace: %s\n", resourceAccessReviewResponse.Namespace)
+	if o.allNamespaces && resourceAccessReviewResponse.Namespace == kapi.NamespaceAll {
```

Comment:
Do we really expect namespace to be different from what we asked?
If no, then I think we don't need namespace comparision check
Otherwise, irrespective of allNamespaces value we need to validate namespace. In case of mismatch, we can show the results as part of the error (at least that will help us in debugging).